unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/Interfaces/EntityIdHelper.cs

```diff
@@ -35,12 +39,57 @@ public long EntityFromLegacyEntityString(string entityId)
                     return (long) SpecialEntityId.THIRD_PERSON_CAMERA_ENTITY_REFERENCE;
             }
 
-            long entityIdLong = entityId.GetHashCode() << 9;
-
+            long entityIdLong = DecodeBase36(entityId) << 9;
+            
             if (!entityIdToLegacyId.ContainsKey(entityIdLong))
                 entityIdToLegacyId[entityIdLong] = entityId;
 
             return entityIdLong;
         }
+        
+        public Int64 DecodeBase36(string input)
+        {
+            string charList = "0123456789abcdefghijklmnopqrstuvwxyz";
+            var reversed = input.ToLower().Reverse();
+            long result = 0;
+            int pos = 0;
+            foreach (char c in reversed)
+            {
+                result += charList.IndexOf(c) * (long)Math.Pow(36, pos);
+                pos++;
+            }
+            return result;
+        }
+        
+        public string ToBase36(long decimalNumber, int radix)
+        {
+            const int bitsInLong = 64;
+            const string digits = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+            if (radix < 2 || radix > digits.Length)
+                throw new ArgumentException("The radix must be >= 2 and <= " + digits.Length.ToString());
+
+            if (decimalNumber == 0)
+                return "0";
+
+            int index = bitsInLong - 1;
+            long currentNumber = Math.Abs(decimalNumber);
+            char[] charArray = new char[bitsInLong];
+
+            while (currentNumber != 0)
+            {
+                int remainder = (int)(currentNumber % radix);
+                charArray[index--] = digits[remainder];
+                currentNumber = currentNumber / radix;
+            }
+
+            string result = new String(charArray, index + 1, bitsInLong - index - 1);
+            if (decimalNumber < 0)
+            {
+                result = "-" + result;
+            }
+
+            return result;
+        }
```

Comment:
This is not being used by anobyd. Is it needed?

Reply:
Yes for Unit testing